src/Modules/OrchardCore.Commerce/Services/CardPaymentService.cs

```diff
@@ -45,7 +47,9 @@ public class CardPaymentService : ICardPaymentService
         IDataProtectionProvider dataProtectionProvider,
         ILogger<CardPaymentService> logger,
         IStringLocalizer<CardPaymentService> stringLocalizer,
-        ISession session)
+        ISession session,
+        IProductInventoryProvider productInventoryProvider,
+        IProductService productService)
```

Comment:
Don't inject `IProductInventoryProvider` (it should be `IEnumerable<IProductInventoryProvider>` anyway). You should use `_productService.UpdateAsync(item)` to update the inventory without having to know which provider is available.

Reply:
Looks like `IProductInventoryService` is not used at all right now. Was this accidentally deleted when you merged from the main branch?

Reply:
As `CardPaymentService` was deleted on dev, this part was moved to `StripePaymentService`. However, instead of the service, I used the providers' `UpdateWithFirstApplicableProviderAsync()` method:
![image](https://user-images.githubusercontent.com/51411356/209790771-58728cba-6eba-4273-8879-7d215d3a791b.png)

Is it wrong to do it this way?

Reply:
You already do that in `ProductInventoryService.UpdateInventoriesAsync()`, so why not call that in `StripePaymentService` as well?